6.0.4 build failure: undefined symbols

## Environment

```sh
# copypaste this script into your shell and replace it with the output
date
r2 -v
iaito -v
uname -ms
```
Fri Oct  3 16:09:04 AEST 2025
radare2 6.0.4 0 @ darwin-arm-64
birth: git.6.0.4 2025-09-29__11:02:22
commit: 6.0.4
options: gpl release -O1 cs:5 cl:2 make
qt.gui.icc: fromIccProfile: Failed to parse description
qt.gui.icc: fromIccProfile: Failed to parse description
iaito 6.0.4
QThreadStorage: entry 1 destroyed before end of thread 0x84540c100
Darwin arm64
## Description
With https://github.com/radareorg/iaito/pull/233 I can no longer build iaito because of error:
```
:info:build /Library/Developer/CommandLineTools/usr/bin/clang++ -stdlib=libc++ -Wl,-rpath, -rdynamic -headerpad_max_install_names  -arch arm64 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk -mmacosx-version-min=26.0 -o iaito.app/Contents/MacOS/iaito  Main.o Iaito.o EditStringDialog.o WriteCommandsDialogs.o DisassemblerGraphView.o OverviewView.o RichTextPainter.o InitialOptionsDialog.o AboutDialog.o CommentsDialog.o EditInstructionDialog.o FlagDialog.o RemoteDebugDialog.o NativeDebugDialog.o XrefsDialog.o MainWindow.o Helpers.o TextEditDialog.o Highlighter.o MdHighlighter.o DirectionalComboBox.o TypeScriptHighlighter.o AnalTask.o AsmOptionsWidget.o FortuneDialog.o NewFileDialog.o CommentsWidget.o ConsoleWidget.o CustomCommandWidget.o Dashboard.o EntrypointWidget.o ExportsWidget.o FlagsWidget.o FunctionsWidget.o ImportsWidget.o CodeEditor.o Omnibar.o RelocsWidget.o SectionsWidget.o SegmentsWidget.o StringsWidget.o SymbolsWidget.o DisassemblyContextMenu.o DecompilerContextMenu.o DisassemblyWidget.o HexdumpWidget.o Configuration.o Colors.o SaveProjectDialog.o TempConfig.o SvgIconEngine.o SyntaxHighlighter.o DecompilerWidget.o VisualNavbar.o GraphView.o PreferencesDialog.o AppearanceOptionsWidget.o GraphOptionsWidget.o PreferenceCategory.o InitializationFileEditor.o QuickFilterView.o ClassesWidget.o ResourcesWidget.o VTablesWidget.o TypesWidget.o HeadersWidget.o SearchWidget.o IaitoApplication.o PythonAPI.o R2PluginsDialog.o IaitoDockWidget.o IaitoTreeWidget.o MapsWidget.o BinariesWidget.o FilesWidget.o GraphWidget.o OverviewWidget.o JsonTreeItem.o JsonModel.o VersionInfoDialog.o ZignaturesWidget.o AsyncTask.o DecompileTask.o AsyncTaskDialog.o StackWidget.o RegistersWidget.o ThreadsWidget.o ProcessesWidget.o BacktraceWidget.o MapFileDialog.o CommandTask.o ProgressIndicator.o R2Task.o R2TaskDialog.o DebugActions.o MemoryMapWidget.o DebugOptionsWidget.o PluginsOptionsWidget.o BreakpointWidget.o BreakpointsDialog.o AttachProcDialog.o RegisterRefsWidget.o SetToDataDialog.o EditVariablesDialog.o EditFunctionDialog.o IaitoTreeView.o ComboQuickFilterView.o HexdumpRangeDialog.o QtResImporter.o IaitoSeekable.o RefreshDeferrer.o WelcomeDialog.o RunScriptTask.o EditMethodDialog.o TypesInteractionDialog.o SdbWidget.o PythonManager.o PluginManager.o BasicBlockHighlighter.o BasicInstructionHighlighter.o LinkTypeDialog.o ColorPicker.o ColorThemeWorker.o ColorThemeComboBox.o ColorThemeListView.o ColorThemeEditDialog.o UpdateWorker.o MemoryDockWidget.o CrashHandler.o BugReporting.o HighDpiPixmap.o GraphGridLayout.o HexWidget.o R2AIWidget.o SelectionHighlight.o Decompiler.o R2GhidraCmdDecompiler.o ShortcutKeys.o ShortcutKeysDialog.o R2pdcCmdDecompiler.o R2DecaiDecompiler.o R2AnotesDecompiler.o R2retdecDecompiler.o AddressableItemContextMenu.o AddressableItemModel.o ListDockWidget.o MultitypeFileSaveDialog.o BoolToggleDelegate.o IOModesController.o SettingsUpgrade.o LayoutManager.o IaitoLayout.o GraphHorizontalAdapter.o ResourcePaths.o IaitoGraphView.o SimpleTextGraphView.o R2GraphWidget.o CallGraph.o AddressableDockWidget.o AnalOptionsWidget.o KeyboardOptionsWidget.o DecompilerHighlighter.o PackageManagerDialog.o qrc_resources.o qrc_native.o qrc_dark.o qrc_midnight.o qrc_light.o moc_Iaito.o moc_EditStringDialog.o moc_WriteCommandsDialogs.o moc_ClickableSvgWidget.o moc_DisassemblerGraphView.o moc_OverviewView.o moc_TypeScriptHighlighter.o moc_ShortcutKeys.o moc_ShortcutKeysDialog.o moc_TextEditDialog.o moc_Highlighter.o moc_MdHighlighter.o moc_DirectionalComboBox.o moc_AnalTask.o moc_AboutDialog.o moc_AsmOptionsWidget.o moc_CommentsDialog.o moc_EditInstructionDialog.o moc_FlagDialog.o moc_RemoteDebugDialog.o moc_NativeDebugDialog.o moc_XrefsDialog.o moc_MainWindow.o moc_InitialOptionsDialog.o moc_NewFileDialog.o moc_CommentsWidget.o moc_ConsoleWidget.o moc_CustomCommandWidget.o moc_Dashboard.o moc_EntrypointWidget.o moc_ExportsWidget.o moc_FlagsWidget.o moc_FunctionsWidget.o moc_ImportsWidget.o moc_Omnibar.o moc_RelocsWidget.o moc_CodeEditor.o moc_SectionsWidget.o moc_SegmentsWidget.o moc_StringsWidget.o moc_SymbolsWidget.o moc_DisassemblyContextMenu.o moc_DecompilerContextMenu.o moc_DisassemblyWidget.o moc_HexdumpWidget.o moc_Configuration.o moc_SaveProjectDialog.o moc_SyntaxHighlighter.o moc_DecompilerWidget.o moc_VisualNavbar.o moc_GraphView.o moc_PreferencesDialog.o moc_AppearanceOptionsWidget.o moc_GraphOptionsWidget.o moc_InitializationFileEditor.o moc_QuickFilterView.o moc_ClassesWidget.o moc_ResourcesWidget.o moc_IaitoApplication.o moc_VTablesWidget.o moc_TypesWidget.o moc_HeadersWidget.o moc_SearchWidget.o moc_R2PluginsDialog.o moc_IaitoDockWidget.o moc_IaitoTreeWidget.o moc_MapsWidget.o moc_BinariesWidget.o moc_FilesWidget.o moc_GraphWidget.o moc_OverviewWidget.o moc_VersionInfoDialog.o moc_ZignaturesWidget.o moc_AsyncTask.o moc_DecompileTask.o moc_AsyncTaskDialog.o moc_StackWidget.o moc_RegistersWidget.o moc_ThreadsWidget.o moc_ProcessesWidget.o moc_BacktraceWidget.o moc_FortuneDialog.o moc_MapFileDialog.o moc_StringsTask.o moc_FunctionsTask.o moc_CommandTask.o moc_R2Task.o moc_R2TaskDialog.o moc_DebugActions.o moc_MemoryMapWidget.o moc_DebugOptionsWidget.o moc_PluginsOptionsWidget.o moc_BreakpointWidget.o moc_BreakpointsDialog.o moc_AttachProcDialog.o moc_RegisterRefsWidget.o moc_SetToDataDialog.o moc_EditVariablesDialog.o moc_EditFunctionDialog.o moc_IaitoTreeView.o moc_ComboQuickFilterView.o moc_HexdumpRangeDialog.o moc_IaitoSeekable.o moc_RefreshDeferrer.o moc_WelcomeDialog.o moc_RunScriptTask.o moc_EditMethodDialog.o moc_TypesInteractionDialog.o moc_SdbWidget.o moc_PythonManager.o moc_PluginManager.o moc_UpdateWorker.o moc_ColorPicker.o moc_ColorThemeWorker.o moc_ColorThemeComboBox.o moc_MemoryDockWidget.o moc_ColorThemeListView.o moc_ColorThemeEditDialog.o moc_KeyboardOptionsWidget.o moc_LinkTypeDialog.o moc_HexWidget.o moc_R2AIWidget.o moc_Decompiler.o moc_R2GhidraCmdDecompiler.o moc_R2DecaiDecompiler.o moc_R2AnotesDecompiler.o moc_R2pdcCmdDecompiler.o moc_R2retdecDecompiler.o moc_AddressableItemContextMenu.o moc_ListDockWidget.o moc_MultitypeFileSaveDialog.o moc_IOModesController.o moc_LayoutManager.o moc_IaitoGraphView.o moc_SimpleTextGraphView.o moc_R2GraphWidget.o moc_CallGraph.o moc_AddressableDockWidget.o moc_AnalOptionsWidget.o moc_DecompilerHighlighter.o moc_PackageManagerDialog.o   -F/opt/local/libexec/qt6/lib -L/opt/local/lib -lr_core -lr_config -lr_debug -lr_bin -lr_lang -lr_anal -lr_bp -lr_egg -lr_asm -lr_arch -lr_esil -lr_flag -lr_reg -lr_search -lr_syscall -lr_fs -lr_io -lr_socket -lr_cons -lr_magic -lr_muta -lr_util -ldl -framework QtSvgWidgets -framework QtWidgets -framework QtSvg -framework QtGui -framework AppKit -framework ImageIO -framework Metal -framework QtNetwork -framework QtCore -framework IOKit -framework DiskArbitration -framework UniformTypeIdentifiers -framework OpenGL   
:info:build Undefined symbols for architecture arm64:
:info:build   "_main", referenced from:
:info:build       <initial-undefines>
:info:build ld: symbol(s) not found for architecture arm64
:info:build clang++: error: linker command failed with exit code 1 (use -v to see invocation)
:info:build make[1]: *** [iaito.app/Contents/MacOS/iaito] Error 1
:info:build make[1]: Leaving directory `/opt/local/var/macports/build/iaito-ad6fe33f/work/iaito-6.0.4/build'
:info:build make: *** [iaito] Error 2
:info:build make: Leaving directory `/opt/local/var/macports/build/iaito-ad6fe33f/work/iaito-6.0.4'
```
The root cause seems to be the incorrect `$$(R2_LIBDIR)` in the project file, which references the environment variable `R2_LIBDIR`. To reference the previously set variable (by `r2 -H R2_LIBDIR`) it should be `$$R2_LIBDIR`.
Another workaround is adding `R2_LIBDIR=/some/dir` build arg when executing `make`.

## Test
Pull the 6.0.4 tarball and build. Full build log from macports attached below.

[main.log](https://github.com/user-attachments/files/22675410/main.log)

--- a/src/Iaito.pro
+++ b/src/Iaito.pro
@@ -21,9 +21,9 @@ unix:!IAITO_BUNDLE_R2_APPBUNDLE {
     isEmpty(R2_LIBDIR) {
         R2_LIBDIR = $$system(r2 -H R2_LIBDIR)
     }
-    QMAKE_RPATHDIR += $$(R2_LIBDIR)
+    QMAKE_RPATHDIR += $${R2_LIBDIR}
     QMAKE_LFLAGS_RPATH=
-    QMAKE_LFLAGS += "-Wl,-rpath,$$(R2_LIBDIR)"
+    QMAKE_LFLAGS += "-Wl,-rpath,$${R2_LIBDIR}"
 }
 
 QMAKE_CXXFLAGS += $$(CXXFLAGS)
@@ -126,13 +126,12 @@ macx {
     QMAKE_INFO_PLIST = macos/Info.plist
 }
 
-unix:exists(/usr/local/include/libr)|bsd:exists(/usr/local/include/libr) {
-    INCLUDEPATH += /usr/local/include/libr
-    INCLUDEPATH += /usr/local/include/libr/sdb
-}
-unix:exists(/usr/include/libr) {
-    INCLUDEPATH += /usr/include/libr
-    INCLUDEPATH += /usr/include/libr/sdb
+unix {
+    isEmpty(R2_INCDIR) {
+        R2_INCDIR = $$system(r2 -H R2_INCDIR)
+    }
+    INCLUDEPATH += $${R2_INCDIR}
+    INCLUDEPATH += $${R2_INCDIR}/sdb
 }
 unix {
     QMAKE_LFLAGS += -rdynamic # Export dynamic symbols for plugins

